### ISSUE TITLE
Scroll to the selected option when clearing filter in editor settings shortcuts

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot-proposals#3020]
### Describe the project you are working on

Irrelevant.

### Describe the problem or limitation you are having in your project

When I search for a shortcut, I sometimes want to look at the nearby shortcuts. But as soon as I remove the filter, the shortcut lists go back to beginning. E.g. here I'd like to see other filesystem shortcuts, so I search one of them, but then this happens
![rzUf48qfAu](https://user-images.githubusercontent.com/2223172/126400945-b866b727-e2a1-4e21-b913-8eacd5e9fe3d.gif)
This is a quite common issue for me. Navigating to a particular editor in such a long list is a bit problematic.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When the filter is cleared, the dialog should remember selected option and scroll to it.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically this https://github.com/godotengine/godot/pull/45812 but for editor shortcuts.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

It's about editor usability.

**Comment by KoBeWi** [comment]
Implemented by https://github.com/godotengine/godot/pull/110690
